nomad/state/state_store.go

```diff
@@ -4498,6 +4514,70 @@ func (s *StateStore) updateJobScalingPolicies(index uint64, job *structs.Job, tx
 	return nil
 }
 
+// updateJobCSIPlugins runs on job update indexes the job in the plugin
+func (s *StateStore) updateJobCSIPlugins(index uint64, job, prev *structs.Job, txn *memdb.Txn) error {
+	ws := memdb.NewWatchSet()
+	plugIns := make(map[string]*structs.CSIPlugin)
+
+	loop := func(job *structs.Job, delete bool) error {
+		for _, tg := range job.TaskGroups {
+			for _, t := range tg.Tasks {
+				if t.CSIPluginConfig == nil {
+					continue
+				}
+
+				plugIn, ok := plugIns[t.CSIPluginConfig.ID]
+				if !ok {
+					p, err := s.CSIPluginByID(ws, t.CSIPluginConfig.ID)
+					if err != nil {
+						return fmt.Errorf("%v", err)
```

Comment:
We can probably just `return err` here and elsewhere in this `updateJobCSIPlugins` method where we're not adding any new context to the error.

Reply:
that's my emacs snippet 🤦